webhook/resources/replica/validator.go

```diff
@@ -106,21 +106,26 @@ func (r *replicaValidator) validateReplicaDeletion(replica *longhorn.Replica) er
 
 	replicas, err := r.ds.ListVolumeReplicasRO(volume.Name)
 	if err != nil {
-		return errors.Wrapf(err, "failed to list replicas for volume %v before deleting replica", volume.Name)
+		return errors.Wrapf(err, "failed to list replicas for volume %v before deleting replica %v", volume.Name, replica.Name)
 	}
 
 	availableReplicas := map[string]struct{}{}
 	for _, r := range replicas {
 		// If the healthyAt as well as failedAt are set to non-empty string,
 		// the replica is still regarded as **available** because its data is probably
 		// intact and can be used for rescue if other replicas are not available anymore.
-		if r.Spec.HealthyAt != "" && r.DeletionTimestamp == nil {
-			availableReplicas[r.Name] = struct{}{}
+		if !datastore.IsAvailableHealthyReplica(r) {
+			continue
+		}
+		if r.Spec.EvictionRequested {
+			continue
 		}
+
+		availableReplicas[r.Name] = struct{}{}
 	}
 	if len(availableReplicas) == 1 {
 		if _, ok := availableReplicas[replica.Name]; ok {
-			return fmt.Errorf("cannot delete replica %v because volume %v only has one available replica", replica.Name, volume.Name)
+			return fmt.Errorf("no other healthy replica available, cannot delete replica %v since it may still contain data for recovery", replica.Name)
```

Comment:
Suggest keeping the volume name to provide better context for debugging.
```suggestion
			return fmt.Errorf("no other healthy replica available for volume %v, cannot delete replica %v since it may still contain data for recovery", volume.Name, replica.Name)
```